src/connector/src/parser/postgres.rs

```diff
@@ -313,3 +350,81 @@ pub fn postgres_row_to_owned_row(row: tokio_postgres::Row, schema: &Schema) -> O
     }
     OwnedRow::new(datums)
 }
+
+fn postgres_decimal_to_string(row: &tokio_postgres::Row, idx: usize, name: &str) -> Option<String> {
+    let res = row.try_get::<_, Option<PgNumeric>>(idx);
+    match res {
+        Ok(val) => {
+            if let Some(PgNumeric {
+                n: Some(big_decimal),
+            }) = val
+            {
+                Some(big_decimal.to_string())
+            } else {
+                None
+            }
+        }
+        Err(err) => {
+            if let Ok(suppressed_count) = LOG_SUPPERSSER.check() {
+                tracing::error!(
+                    column = name,
+                    error = %err.as_report(),
+                    suppressed_count,
+                    "parse column failed",
+                );
+            }
+            None
+        }
+    }
+}
+
+fn postgres_decimal_to_rw_int256(
+    row: &tokio_postgres::Row,
+    idx: usize,
+    name: &str,
+    builder: Option<&mut ArrayBuilderImpl>,
+) -> Option<ScalarImpl> {
+    // Currently in order to handle the decimal beyond RustDecimal,
+    // we use the PgNumeric type to convert the decimal to a string.
+    // Then we convert the string to Int256.
+    let string = postgres_decimal_to_string(row, idx, name)?;
+    match Int256::from_str(string.as_str()) {
+        Ok(num) => {
+            if let Some(builder) = builder {
+                builder.append(Some(ScalarImpl::from(num)));
+                None
+            } else {
+                Some(ScalarImpl::from(num))
```

Comment:
This API feels wired to me. Why not always return `ScalarImpl` and update array builder outside?

Reply:
refactored.